java-client/src/main/java/org/opensearch/client/opensearch/OpenSearchHttpClient.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.client.opensearch;
+
+import java.io.IOException;
+import java.util.function.Function;
+
+import javax.annotation.Nullable;
+
+import org.opensearch.client.ApiClient;
+import org.opensearch.client.Response;
```

Comment:
This class comes from `RestClient` which we should not have (it would be acceptable for a `RestClientTransport` but not for the `OpenSearchHttpClient` which should stay transport neutral.